### ISSUE TITLE
Problems using UDP influxdb output

### ISSUE THREAD

**Issue by mcfedr** [influxdata/telegraf#2862]
Looks like telegraf is splitting a udp message and this means its not getter parsed properly by influxdb.

This has started happening since update to telegraf 1.3

### Relevant telegraf.conf:

```toml
[[outputs.influxdb]]
  urls = ["udp://influxdb:8089"]
  database = "telegraf"
  retention_policy = ""
  write_consistency = "any"
  timeout = "5s

[[inputs.system]]
```

### System info:

Telegraf 1.3 docker image
Influxdb 1.2 docker image

### Log message

```
influxdb_1    | [I] 2017-05-29T09:25:00Z Failed to parse points: unable to parse 'system,host=proxy-docker uptime_format="5 days 1496049900000000000': unbalanced quotes service=udp
influxdb_1    | [I] 2017-05-29T09:25:00Z Failed to parse points: unable to parse 'system,host=proxy-docker   1:26" 1496049900000000000': invalid field format service=udp
```

**Comment by oplehto** [comment]
I am seeing something which may be related with the socket writer plugin with UDP: Unrelated metrics are getting randomly dropped when there are measurements active which require a split into multiple UDP packets.

I haven't had time to look in any deeper but I suspect that this loop breaks prematurely: https://github.com/influxdata/telegraf/blob/f74687dcc0ceb128027414c7099f7c70df86d1ad/plugins/outputs/influxdb/client/udp.go#L75

**Comment by sebito91** [comment]
The splitter was updated before the v1.3.0 release here: https://github.com/influxdata/telegraf/pull/2795. Not sure if we're lopping off a value too quickly?

**Comment by danielnelson** [comment]
I think maybe we shouldn't determine the buffer length until after splitting https://github.com/influxdata/telegraf/blob/master/plugins/outputs/influxdb/influxdb.go#L189

**Comment by sebito91** [comment]
I see what happened, it was refactored slightly on #2799 since we merged #2795! @danielnelson, didn't realize we actually merged that...I think we built with just #2795 but will confirm in the morning.

**Comment by danielnelson** [comment]
I'm pretty sure this is it, working on PR now.

**Comment by danielnelson** [comment]
The invalid points was fixed in 1.3.1, but there was still an edge case where long point are not split correctly and they would be dropped.  This is fixed in 1.3.2.

**Comment by mcfedr** [comment]
Deployed 1.3.1 so far, not seeing the issues anymore
